src/vcpkg/vcpkgpaths.cpp

```diff
@@ -834,6 +834,11 @@ namespace vcpkg
             .map([](Path&& dot_git_parent) { return std::move(dot_git_parent) / ".git"; });
     }
 
+    bool VcpkgPaths::is_overlay_port(const std::string& port_name) const
+    {
+        return Util::contains(overlay_ports, port_name);
```

Comment:
I'm pretty sure overlay_ports is a list of directories and thus this change won't behave correctly. Can you add tests for this?

Reply:
I'll check that. At least the name and type suggests that these are in fact port names.

Reply:
> At least the name and type suggests that these are in fact port names.

I believe the name is based on the name of the switch being `--overlay-ports`.

> Do you mean unit tests or e2e tests?

Of course unit tests are better but I doubt we have great ways to do that.